qiime2/sdk/usage.py

```diff
@@ -77,8 +77,10 @@ def validate(self, signature):
 
         extra = provided - exp_inputs - exp_params
         if len(extra) > 0:
-            raise ValueError('Extra input(s) or parameter(s): %r' %
-                             (extra, ))
+            for provided_value in extra:
```

Comment:
@David-Rod - I suspect you won't need any of this once https://github.com/qiime2/qiime2/pull/530 is merged.